distribution/pull_v2_windows.go

```diff
@@ -133,21 +132,43 @@ func (mbv manifestsByVersion) Swap(i, j int) {
 	mbv.list[i], mbv.list[j] = mbv.list[j], mbv.list[i]
 }
 
-// checkImageCompatibility blocks pulling incompatible images based on a later OS build
+// checkImageCompatibility tries to check whether a Windows image is incompatible
+// with the host's OS version. It is a no-op for non-Windows images. It returns
+// an error if the image is incompatible, but ignores failures when parsing
+// imageOSVersion.
+//
 // Fixes https://github.com/moby/moby/issues/36184.
 func checkImageCompatibility(imageOS, imageOSVersion string) error {
-	if imageOS == "windows" {
-		hostOSV := osversion.Get()
-		splitImageOSVersion := strings.Split(imageOSVersion, ".") // eg 10.0.16299.nnnn
-		if len(splitImageOSVersion) >= 3 {
-			if imageOSBuild, err := strconv.Atoi(splitImageOSVersion[2]); err == nil {
-				if imageOSBuild > int(hostOSV.Build) {
-					errMsg := fmt.Sprintf("a Windows version %s.%s.%s-based image is incompatible with a %s host", splitImageOSVersion[0], splitImageOSVersion[1], splitImageOSVersion[2], hostOSV.ToString())
-					log.G(context.TODO()).Debugf(errMsg)
-					return errors.New(errMsg)
-				}
-			}
-		}
+	if imageOS != "windows" {
+		return nil
+	}
+
+	splitImageOSVersion := strings.SplitN(imageOSVersion, ".", 4) // eg 10.0.16299.nnnn
+	if len(splitImageOSVersion) < 3 {
+		return nil
+	}
+	imgMajor, err := strconv.ParseUint(splitImageOSVersion[0], 10, 8)
+	if err != nil {
+		return nil
+	}
+	imgMinor, err := strconv.ParseUint(splitImageOSVersion[1], 10, 8)
+	if err != nil {
+		return nil
+	}
+	imgBuild, err := strconv.ParseUint(splitImageOSVersion[2], 10, 8)
```

Comment:
```suggestion
	imgBuild, err := strconv.ParseUint(splitImageOSVersion[2], 10, 16)
```

build version is 16 bit

Reply:
doh! good catch; copy/pasta on my side

Reply:
fixed, and added a basic unit-test as well 😄